src/main/java/roomescape/common/controller/CommonController.java

```diff
@@ -0,0 +1,18 @@
+package roomescape.common.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class CommonController {
+
+    @GetMapping("/reservation")
+    public String getReservationPage() {
+        return "reservation";
+    }
+
+    @GetMapping("/")
```

Comment:
```suggestion
    @GetMapping
```
"/" 는 따로 작성하지 않아도 됩니다. ㅎㅎ

Reply:
붙이지 않은 것과 "/"를 붙인 것이 정말 같은 의미인 지 테스트해보기 위해서
1. `GetMapping`
2. `GetMapping("")`
3. `GetMapping("/")` 
로 나누어 실험해보았는데요, 

1번과 2번은 공존할 경우 요청을 보내기 이전부터 중복 메서드로 인한 충돌이 발생합니다.

```
Caused by: java.lang.IllegalStateException: Ambiguous mapping. Cannot map 'commonController' method 
roomescape.common.controller.CommonController#getPopularThemePage3()
to {GET [/ || ]}: There is already 'commonController' bean method
```

다만 1,2번 중 하나와 3번이 동시에 존재할 경우 요청을 보낼 때에 에러가 발생합니다.

```
java.lang.IllegalStateException: Ambiguous handler methods mapped for '/': {public java.lang.String roomescape.common.controller.CommonController.getPopularThemePage(), public java.lang.String roomescape.common.controller.CommonController.getPopularThemePage4()}
```

1, 2, 3번 모두 `localhost:8080`, `localhost:8080/` 으로의 GET 요청을 핸들링하는데, Spring이 "/"이 붙은 요청과 안 붙은 요청을 동일하게 취급하는 것이 맞다면, 왜 1 or 2번과 3번이 동시에 존재하는 경우에, 서버 실행 시에(요청을 보내기 이전)에는 에러를 발생시키지 않을까요?

제가 생각한 이유는 
1. 빈 생성 시에는 동일한 요청 uri를 가진 메서드가 존재하는 지만 체크한다. ("", "/"를 다른 메서드로 인식한다.)
2. 요청이 왔을 때 필요시 "/"를 추가/제거해주는 로직이 존재한다.
3. 요청을 어느 곳으로 매핑해주어야할 지 판단을 할 수 없어 에러를 발생시킨다.

인데, 라라는 이에 대해 어떻게 생각하시나요? 제가 모르는 어떤 부분이 있는 걸까요? 
내부 작동 로직이 중요하다고 생각하지 않지만, 단순히 궁금증이 생겨 질문드립니다!

Reply:
우선 `@GetMapping`이 간결하다고 생각해서 코드는 그렇게 수정했습니다!

Reply:
1. 매핑 등록 시점의 URI 구분
- Spring은 내부적으로 RequestMappingInfo를 통해 매핑 정보를 관리하는데,
일단 `@RequestMapping("")`과 `@RequestMapping("/")`는 Java 코드 수준에서는 문자열이 다르므로 다른 매핑으로 간주됩니다. ex) `/api/reservation/` 와 `/api/reservation` 는 문자열 기준으로 다른 URI로 인식

2. 요청 처리 시점의 경로 정규화
- 저도 이부분은 정확히 알지 못했던 부분이라 찾아보았는데, 
HTTP 요청이 들어오면 Spring은 경로를 정규화하여 처리하기때문에,  `/`, `""`가 동일한 경로로 해석될 수 있다고 하네요. 🤔
그래서 매핑된 핸들러가 둘 이상 존재할 경우 발생하는 Ambiguous handler methods 예외가 발생하게 되는 것 같아요.

매핑 등록 시점에는 문자열 비교만 하고, 요청 매칭 시점에는 경로 정규화를 거치기 때문에 중복 핸들러가 뒤늦게 감지되는 것 같네요. ㅎㅎ

Reply:
이 내용을 적어두고, 나중에 1과 2를 구분한 이유에 대해 생각해보는 것도 좋은 학습이 될 것 같다는 생각이 드네요! 자세히 설명해주셔서 감사합니다. ☺️